src/analysis/complex/real_deriv.lean

```diff
@@ -62,3 +61,115 @@ times_cont_diff_iff_times_cont_diff_at.2 $ λ x,
   h.times_cont_diff_at.real_of_complex
 
 end real_deriv_of_complex
+
+section complex_fderiv_properties
+/-! ### Antiholomorphy and conformality of complex functions -/
+open complex continuous_linear_map
+
+variables {E : Type*} [normed_group E] [normed_space ℝ E]
+  {z : ℂ} {f : ℂ → E}
+
+lemma has_fderiv_at_conj (z : ℂ) : has_fderiv_at conj conj_cle.to_continuous_linear_map z :=
+conj_cle.has_fderiv_at
+
+lemma fderiv_conj_eq_conj_fderiv {z : ℂ} (h : differentiable_at ℝ f z) :
+  (fderiv ℝ f z).comp conj_cle.to_continuous_linear_map = fderiv ℝ (f ∘ conj) (conj z) :=
+begin
+  rw ← conj_conj z at h,
+  let p := fderiv.comp (conj z) h (has_fderiv_at_conj $ conj z).differentiable_at,
+  rw [conj_conj, (has_fderiv_at_conj $ conj z).fderiv] at p,
+  exact p.symm,
+end
+
+/-- A (real-differentiable) complex function `f` is antiholomorphic if and only if there exists some
+    complex linear map `g'` that equals to the composition of `f`'s differential and the conjugate
+    function -/
+lemma antiholomorph_at_iff_exists_complex_linear_conj
+  [normed_space ℂ E] [is_scalar_tower ℝ ℂ E]
+  (hf : differentiable_at ℝ f z) : differentiable_at ℂ (f ∘ conj) (conj z) ↔
+  ∃ (g' : ℂ →L[ℂ] E), g'.restrict_scalars ℝ =
+  (fderiv ℝ f z).comp conj_cle.to_continuous_linear_map :=
+begin
+  split,
+  { intros h,
+    rw ← conj_conj z at hf,
+    rcases (differentiable_at_iff_exists_linear_map ℝ $
+      hf.comp (conj z) (has_fderiv_at_conj $ conj z).differentiable_at).mp h with ⟨f', hf'⟩,
+    rw conj_conj at hf,
+    rw ← fderiv_conj_eq_conj_fderiv hf at hf',
+    exact ⟨f', hf'⟩, },
+  { rintros ⟨g', hg'⟩,
+    rw ← conj_conj z at hf hg',
+    exact ⟨g', has_fderiv_at_of_eq ℝ
+      (hf.has_fderiv_at.comp (conj z) $ has_fderiv_at_conj $ conj z) hg'⟩, },
+end
+
+end complex_fderiv_properties
+
+open complex continuous_linear_map
+
+section conformal_complex_normed_space
+
+variables {E : Type*} [normed_group E] [normed_space ℝ E] [normed_space ℂ E]
+  [is_scalar_tower ℝ ℂ E] {z : ℂ} {f : ℂ → E}
+
+/-- A real differentiable function of the complex plane into some complex normed space `E` is
+    conformal at a point `z` if it is holomorphic or antiholomorphic at that point -/
+lemma conformal_at_of_holomorph_or_antiholomorph_at
+  (hf : differentiable_at ℝ f z) (hf' : fderiv ℝ f z ≠ 0)
+  (h : differentiable_at ℂ f z ∨ differentiable_at ℂ (f ∘ conj) (conj z)) :
```

Comment:
This would be more useful as two separate lemmas than with the `∨` hypothesis; can you please split it?  Also, I think `hf` is unnecessary.